### PR TITLE
react-grid-layout Layout Interface Update

### PR DESCRIPTION
**Pull request by ZheyangSong**
+ Adjust `Layout` to match the source code's defintion

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/STRML/react-grid-layout/blob/master/lib/utils.js#L5
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by RyanCavanaugh** [comment]
@ZheyangSong Please fix the failures indicated in the Travis CI log.

**Comment by ZheyangSong** [comment]
@RyanCavanaugh Removed patch segment in version number. It should fix the CI.

File: types/react-grid-layout/index.d.ts
Comment (alitaheri) [inline types/react-grid-layout/index.d.ts]: Why? Is this required now?

**Review by alitaheri — CHANGES_REQUESTED** [review]
If you got the time add `isRearrangeable` thanks 👍

File: types/react-grid-layout/index.d.ts
Comment (ZheyangSong) [reply types/react-grid-layout/index.d.ts]: @alitaheri yes, it's required. Please refer to the URL in the PR summary. More details can be found in the library's source repo...

Comment (ZheyangSong) [reply types/react-grid-layout/index.d.ts]: @alitaheri Alright, misled by the flow type definition in the source code. Will make it optional and add the `isRearrangeable`.

**Comment by RyanCavanaugh** [comment]
@alitaheri - Thanks for your review of this PR! Can you please look at the new code and update your review status if appropriate?

**Review by alitaheri — APPROVED** [review]
LGTM 👍 

Thanks

**Comment by RyanCavanaugh** [comment]
Approved by a listed owner. PR ready to merge pending express review by a maintainer.

**Pull request merged**
